flask_appbuilder/security/sqla/manager.py

```diff
@@ -304,9 +304,18 @@ def update_role(self, pk, name: str) -> Optional[Role]:
             return
 
     def find_role(self, name):
-        return (
+        role = (
             self.get_session.query(self.role_model).filter_by(name=name).one_or_none()
         )
+        if role:
+            return role
+        elif self.auth_partial_matching:
```

Comment:
this seems a bit dangerous, since this is a public method that can be used by downstream methods, so fetching a role would change depending on `auth_partial_matching` is unexpected. Also doesn't the like method need wildcards?

Reply:
You are correct about the wildcards I should add %{}% for using like method.
You review gave point to think about and what I did was wrong.
Now I changed my logic and move it the relevant function.